cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/types/TypeParser.java

```diff
@@ -47,75 +48,23 @@ public class TypeParser {
   private static final Logger log = LoggerFactory.getLogger(TypeParser.class);
 
   public static final String UNKNOWN_TYPE_STRING = "UNKNOWN";
-  public static final List<String> PRIMITIVES =
-      List.of(
-          "byte",
-          "short",
-          "int",
-          "long",
-          "float",
-          "double",
-          "boolean",
-          "char",
-          // LLVM primitive types
-          "i1",
-          "i8",
-          "i32",
-          "i64",
-          "i128",
-          "half",
-          "bfloat",
-          "fp128",
-          "x86_fp80",
-          "ppc_fp128");
+
   private static final Pattern functionPtrRegex =
       Pattern.compile(
           "(?:(?<functionptr>[\\h(]+[a-zA-Z0-9_$.<>:]*\\*\\h*[a-zA-Z0-9_$.<>:]*[\\h)]+)\\h*)(?<args>\\(+[a-zA-Z0-9_$.<>,\\*\\&\\h]*\\))");
```

Comment:
What is this? Shouldn't this be a Language trait? And some documentation would be nice. regex101.com can't parse this regex...

Reply:
Good question. There's also a regex in the `TypeManager` (`funPointerPattern`) which looks quite different

Reply:
I have added a TODO to this, let's overhaul this when we overhaul the type system.